sdk/src/main/java/io/customer/sdk/di/CustomerIOComponent.kt

```diff
@@ -90,11 +113,14 @@ internal class CustomerIOComponent(
         }
     }
 
+    fun buildMoshi(): Moshi = Moshi.Builder()
```

Comment:
should this be `lazy` or would we require to build a new instance?

Reply:
As long as we don't make it a singleton (since it doesn't need to be), I am fine with using lazy or not. If we make the `moshi` property a `lazy`, that makes me want to ask the question, "Why not make everything `lazy`?" and to that question, I don't know if making lots of properties `lazy` is a good thing?

Reply:
> "Why not make everything lazy?"

Lazy initialization was designed to prevent the unnecessary initialization of objects and it comes down to how expensive is the `initialization`. 

Moshi does a maybe few milliseconds of reflection each time it creates a JSON adapter for a type it's never before seen. If you create many Moshi instances you potentially duplicate this work. This is what we can saved by having it `lazy`. 

The memory issue is not what I am not concerned with much, as its scope is bounded by `CustomerIOComponent`. But if you are comfortable with doing multiple initializations until we notice this performance hit, we can do that as well.